nxontology_ml/tests/test_resources/metal_nxo_features.json

```diff
@@ -1,122 +1,106 @@
     "intrinsic_ic_sanchez": 0.0,
     "intrinsic_ic_sanchez_scaled": 0.0,
-    "n_parents": 0,
-    "n_children": 2,
-    "n_roots": 1,
-    "n_leaves": 5
+    "n_parents": 0.0,
+    "n_roots": 1.0,
+    "n_children": 2.0,
+    "n_leaves": 5.0
   },
   {
-    "identifier": null,
-    "name": "palladium",
-    "depth": 2,
-    "n_ancestors": 3,
-    "n_descendants": 1,
-    "intrinsic_ic": 2.0794415417,
+    "depth": 2.0,
+    "n_ancestors": 3.0,
+    "n_descendants": 1.0,
+    "intrinsic_ic": 2.0794415416798357,
     "intrinsic_ic_scaled": 1.0,
-    "intrinsic_ic_sanchez": 1.5040773968,
-    "intrinsic_ic_sanchez_scaled": 0.8394415783,
-    "n_parents": 1,
-    "n_children": 0,
-    "n_roots": 1,
-    "n_leaves": 1
+    "intrinsic_ic_sanchez": 1.5040773967762742,
+    "intrinsic_ic_sanchez_scaled": 0.8394415782963753,
+    "n_parents": 1.0,
+    "n_roots": 1.0,
+    "n_children": 0.0,
+    "n_leaves": 1.0
   },
   {
-    "identifier": null,
-    "name": "platinum",
-    "depth": 2,
-    "n_ancestors": 3,
-    "n_descendants": 1,
-    "intrinsic_ic": 2.0794415417,
+    "depth": 2.0,
+    "n_ancestors": 3.0,
+    "n_descendants": 1.0,
+    "intrinsic_ic": 2.0794415416798357,
     "intrinsic_ic_scaled": 1.0,
-    "intrinsic_ic_sanchez": 1.5040773968,
-    "intrinsic_ic_sanchez_scaled": 0.8394415783,
-    "n_parents": 1,
-    "n_children": 0,
-    "n_roots": 1,
-    "n_leaves": 1
+    "intrinsic_ic_sanchez": 1.5040773967762742,
+    "intrinsic_ic_sanchez_scaled": 0.8394415782963753,
+    "n_parents": 1.0,
+    "n_roots": 1.0,
+    "n_children": 0.0,
+    "n_leaves": 1.0
   },
   {
-    "identifier": null,
-    "name": "precious",
-    "depth": 1,
-    "n_ancestors": 2,
-    "n_descendants": 5,
-    "intrinsic_ic": 0.4700036292,
-    "intrinsic_ic_scaled": 0.2260239684,
-    "intrinsic_ic_sanchez": 0.6931471806,
-    "intrinsic_ic_sanchez_scaled": 0.3868528072,
-    "n_parents": 1,
-    "n_children": 4,
-    "n_roots": 1,
-    "n_leaves": 4
+    "depth": 1.0,
+    "n_ancestors": 2.0,
+    "n_descendants": 5.0,
+    "intrinsic_ic": 0.47000362924573547,
+    "intrinsic_ic_scaled": 0.2260239683708792,
+    "intrinsic_ic_sanchez": 0.6931471805599453,
+    "intrinsic_ic_sanchez_scaled": 0.3868528072345416,
+    "n_parents": 1.0,
+    "n_roots": 1.0,
+    "n_children": 4.0,
+    "n_leaves": 4.0
   },
   {
-    "identifier": null,
-    "name": "silver",
-    "depth": 2,
-    "n_ancestors": 4,
-    "n_descendants": 1,
-    "intrinsic_ic": 2.0794415417,
+    "depth": 2.0,
+    "n_ancestors": 4.0,
+    "n_descendants": 1.0,
+    "intrinsic_ic": 2.0794415416798357,
     "intrinsic_ic_scaled": 1.0,
-    "intrinsic_ic_sanchez": 1.5686159179,
-    "intrinsic_ic_sanchez_scaled": 0.8754612128,
-    "n_parents": 2,
-    "n_children": 0,
-    "n_roots": 1,
-    "n_leaves": 1
+    "intrinsic_ic_sanchez": 1.5686159179138452,
+    "intrinsic_ic_sanchez_scaled": 0.875461212765156,
+    "n_parents": 2.0,
+    "n_roots": 1.0,
+    "n_children": 0.0,
+    "n_leaves": 1.0
```

Comment:
The float conversion of int fields is irksome, but fine here in the context of features that will be used for an sklearn pipeline.

Reply:
Indeed it's a little sad. The current way I wrote it, all the numerical values are shoved into a numpy array and casted to floats. I could alter the logic to treat ints vs floats differently but not sure if worth it since, as you noted, the model will consume floats at the end of the day. WDYT?

Reply:
Not worth separating ints and floats into separate arrays.